sample/rules.go

```diff
@@ -126,40 +135,53 @@ func (s *RulesBasedSampler) GetSampleRate(trace *types.Trace) (rate uint, keep b
 
 func ruleMatchesTrace(t *types.Trace, rule *config.RulesBasedSamplerRule, checkNestedFields bool) bool {
 	// We treat a rule with no conditions as a match.
-	if rule.Condition == nil {
+	if rule.Conditions == nil {
 		return true
 	}
 
 	var matched int
 
-	for _, condition := range rule.Condition {
+	for _, condition := range rule.Conditions {
 	span:
 		for _, span := range t.GetSpans() {
 			value, exists := extractValueFromSpan(span, condition, checkNestedFields)
-
-			if conditionMatchesValue(condition, value, exists) {
+			if condition.Matches == nil {
```

Comment:
Under what conditions will `condition.Matches` be nil?  If the user doesn't supply a `Datatype` does `.Matches` get a default match function or is it nil?

Reply:
Yeah, if there is not a datatype specified, then the matches function is nil.